Drawsana/Shapes/Implementations/PenShape.swift

```diff
@@ -9,14 +9,18 @@
 import CoreGraphics
 import UIKit
 
-public struct PenLineSegment: Codable {
-  var a: CGPoint
-  var b: CGPoint
-  var width: CGFloat
+public struct PenLineSegment: Codable, Equatable {
+  public var a: CGPoint
+  public var b: CGPoint
+  public var width: CGFloat
 
-  var midPoint: CGPoint {
+  public var midPoint: CGPoint {
     return CGPoint(x: (a.x + b.x) / 2, y: (a.y + b.y) / 2)
   }
+
+  public static func ==(_ a: PenLineSegment, _ b: PenLineSegment) -> Bool {
```

Comment:
Is this implementation necessary? Since Swift 4.1 `Equatable` conformance is automatically generated if possible. Refer to https://github.com/apple/swift-evolution/blob/master/proposals/0185-synthesize-equatable-hashable.md